atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueCache.java

```diff
@@ -31,10 +32,16 @@
  * the central cache (taking in to account the since-locked descriptors), as well as checking for conflicts for
  * serializable transactions by adding a check in the {@link com.palantir.atlasdb.transaction.api.PreCommitCondition}.
  */
-public interface LockWatchValueCache {
+public interface LockWatchValueCache extends ValueCacheUpdater {
+    @Override
     void processStartTransactions(Set<Long> startTimestamps);
 
-    void updateCacheOnCommit(ValueDigest digest, long startTs);
+    @Override
+    void updateCacheOnCommit(Set<Long> startTimestamps);
+
+    default void processFailedTransactions(Set<Long> startTimestamp) {
```

Comment:
hmm, I suppose we need to remove the state if the transaction failed? I can implement this in a subsequent PR as you mentioned.